### ISSUE TITLE
I witnessed something like this IRL

### ISSUE THREAD

**Issue by v32itas** [primaryobjects/colorbot#1]
I witnessed something like this IRL. Wasn't sure until now, but I was studying about this whole night mostly about mongodb and nodejs. Most of it's evidences got destroyed, but there are leftovers of pictures and wordlists collection, but on top of that as I understand same 'neural network' was running 5 instances of modified chrome in background and disguising it's activity under simple adware, while it's really more than that. Those logs were even commented by it, every single step, about 'preffering UAC dissabled' and comparing it to zero digit. I can't event put it to words.. Gonna try to extract that data now

**Comment by cauerego** [comment]
I always get people telling me I write in cryptographic mode... Now to me, that's you, @v32itas! :D

Are you saying you've "encountered" a "virus ai" on your machine doing computer vision tasks? If so, all I can say is I had one of those "encounters" with a "wild ai" myself, although much more [evident](http://talk.cregox.com/t/t-mobile-jessica-chatbot/7738)

**Comment by primaryobjects** [comment]
I'm not really sure what this comment is about. Is it related to this project? :)

**Comment by cauerego** [comment]
@primaryobjects I doubt it! ;P

I ended up here because I followed your project while searching for almost anything about ai, after I started researching for basiux.org

By the way, did you know google just released their Ai? Look for TensorFlow.

**Comment by primaryobjects** [comment]
@cauerego Yep, I've heard about it, thanks. Nice web site, btw. You might be interested in my [AI Programmer](https://github.com/primaryobjects/AI-Programmer) project as well. [Using Artificial Intelligence to Write Self-Modifying/Improving Programs](http://www.primaryobjects.com/CMS/Article149)

**Comment by v32itas** [comment]
[That log](https://github.com/v32itas/systems-thinking/blob/master/win7_forensics.log) which I promised, thank you for links. Not sure how to explain this, but that neural network  was learning about wtf user likes as well as learning local language and pwning win7 security like a boss. But it was collecting images by what user likes, that behavior seemed strange to me, but when I discovered this repository I realized why it was collecting those images.. It's just nice. Started learning about these neural networks few days ago. And this one which I discovered is something that I never believed existing.

**Comment by v32itas** [comment]
I read that log carefully from beginning to the end. And thing is that all log is honest. and my favorite quote would be 

> MSI (s) (A0:3C) [16:46:20:646]: Hello, I'm your 32bit Impersonated custom action server.

**Comment by cauerego** [comment]
@primaryobjects LOL, proof of concept using BrainFuck!! Completely insane! :dancer: Maybe you should be using Piet instead, heard of it? I'm honestly not all that much interested in that project (for future development, cuz I might still read it for reference), but I find it amazing you could build it! EXTRAORDINARY indeed. :+1: 

I would **love to _talk_ more** with you about Ai in general, if you will. There are plenty channels we could use ([I **love** github](http://talk.cregox.com/t/dont-know-why-i-cant-stop-applying-for-jobs/7661/4?u=cregox), but not so much the [issues tracker](https://meta.discourse.org/t/where-is-the-github-issue-tracker/13865?u=cawas)) branching out from www.basiux.org and, currently, there's also a more dynamic channel I'm participating at [slack codenewbie#ai](https://codenewbie.slack.com/messages/ai/)

But if you rather continue here, my 2 questions to you is: how long do you think [Seth Bling took to build Mar i/o](http://talk.cregox.com/t/thanks-to-mari-o-time-to-focus-on-ai/7653?u=cregox) ?  And how long do you think it would take [me to port it to javascript](http://talk.cregox.com/t/thanks-to-mari-o-time-to-focus-on-ai/7653/5?u=cregox) (considering [I'm not such a regular dude](http://talk.cregox.com/t/my-social-unfair-undervaluation-and-why-the-financial-system-is-broken/7751?u=cregox)) ?

And if you're looking for a "next project", please do consider making **mariox**! :)

**Comment by primaryobjects** [comment]
@v32itas Unfortunately, that log message isn't related to anything AI. It comes from a Windows installer (MS Install MSI). See [here](http://stackoverflow.com/questions/16240786/wix-custom-actions-in-64-bit) and [here](http://www.experts-exchange.com/OS/Microsoft_Operating_Systems/Server/Q_27818732.html).

**Comment by cauerego** [comment]
lol. awesome yet again. :100: 

and I had ["opened an issue"](https://github.com/v32itas/systems-thinking/issues/2) to try and talk more about it, because I couldn't figure it out just looking at it... should have thought of googling it. (and I even knew about MSI being Microsoft's Installer)

**Comment by cauerego** [comment]
@primaryobjects how long did you take to make [all](http://www.primaryobjects.com/CMS/Article150) [that](http://www.primaryobjects.com/CMS/Article163)? I can see the repo had movement for at least 2 years! o_O but that means quite little into actual time developing. And what were your main challenges, exactly?

**Comment by v32itas** [comment]
Of course, lets just ignore that whole log is obvious confession of compromise, clearly disabling automatic updates, restarts, shutdowns, installing a TON of adware and new registers, hijacking original MSI session, removing it and overriding signatures for modified adware, taking absolutely full control over machine and disabling all possible notices about that something is wrong. And it did that based on MACHINE POLICY, which had nothing to do with user policy

And this happened only in the very beginning of it's legacy. After that it was successfully running 5 modified chrome browser instances in background and improving the looks of user interface, just however FAILING badly with file and directory names, but it was obsessed with collecting of icon images, there were tons of icons and that NLP nodes were exchanging browser usage information in clear text JSON. Was often compiling. All of this looked mysterious to me, but

5 days ago I started learning JavaScript and neural networks and all funny parts of custom elevated action servers running in user browser, with ability tu run compilers, it's own command line. Yeah if nobody said straight that it is RAT with smart scripts for autoexploitation and perfect error handling, then it's legit right ?.. After a while i discovered this 
https://github.com/v32itas/nlp_compromise it's worth to mention that this NLP was learning users local languages, which is something like polymorphic malware engine, improving it's cloaking on it's own by injecting it's morphed code into existing important system exes as well as morphing tools like 'google chrome' into differently looking but very similar ones and exploiting everything based on some kind of (malicious)MACHINE POLICY which is doubtfully self-learned.

But now I'm pretty sure it was NLP. So yes @primaryobjects it had nothing to do with AI ir was just NLP.

P.S. Just to mention, both URL's of yours from unofficial sources + both about malware "silent instalations" "remote impersonated custom action server"

**Comment by cauerego** [comment]
@v32itas his feedback was completely legitimate, you shouldn't take so much offense in it. Read again your posts, try to see how confusing you sound. It surprises me he even gave it any attention!

You also give no information on who you are or what you're doing, so even if he wanted, it's hard to verify any of your work done. Saying you're researching about JavaScript and neural networks for 5 days (maybe 7 days now, you've made your first contributions 7 days ago) don't give you _any_ credit at all. We both have been at both those topics for years, and they are hard as hell. Granted, myself have only being paying conscious attention at my Ai research since July, but it's quite dedicated and not "5 days", and on top of 30+ years of hands on computer hacking experience.

Let's head forward to the now-back-to-be-still-quite-exciting-offtopic-topic...

What practical signs of this so called Ai have you seen other than this way too big log? What made you stop to look at it in the first place?

And, trying to lighten up the mood...

Loved how you created repos to share your findings! :+1:

**Comment by v32itas** [comment]
Well I'm not a programmer and I don't really understand github, so I fork everything I need. I never judged anyone here for code. However I know my way in shell and I quit using definition AI and started using NLP. During my "sysadmin" days I was always dreaming about such 'swarm intelligence' prototypes. I'm terrible with maths so programming is not really for me. But the thing is that you can code for many years and master programming. But building neural networks is clearly from sysadmins field so after a lot of webapp reversing by trial-and-error. And this dedication which I put into "Systems Thinking" is at 'no life' level for a long time. And It's 7 days in web related stuff not in systems administration and forensics. And I already know how Neural Networks are made, now to make them self-learn is entirely not my area that's why I'm no longer declaring it as AI I call my current study field 'Neural Networks' and learning JS lead me to facebook app developement, which lead to discovery about how ingeniously flexible facebook is and how everyone is improving it every day, facebook is huge neural network.

Ok, so only reason acted offensive previously because that log was ignored, if you dont have time to analyze that it is more appropriate to stay neutral.

@cauerego  not sure why you have deleted message to me, but you can find me https://www.facebook.com/docendo.discimus.1 and there is public PGP key in public profile. As for email you might be able to combine facebook profile url into email  docendo.discimus.1@facebook.com but facebook is really flexible and ever evolving so not sure if this works now. But I prefer keeping knowledge open.

**Comment by cauerego** [comment]
@v32itas I haven't deleted anything. I hardly let my poop go away - [I _hate_ deleting](http://talk.cregox.com/t/from-gods-debris-to-data-loss-why-people-love-forgetting-so-much/7750/1). No idea what you meant there.

I've got 2 devops brothers and 1 programmer in the family. I don't think any of the 2 sysadmins would agree with you that "building neural networks is clearly _from_ sysadmins" if you meant it's something a non-coder would create. Really, really unlikely. Artificial neural networks in the Ai sense is Ai and is about "engineering" code. I don't think you're grasping the whole Ai concept very well there but - then again - I could be wrong. Maybe it already became an archetype and I didn't know. I do know some Ai researchers don't quite realize this, among a lot of important stuff themselves.

All big companies today are a "huge neural network" in the sense you talk, they are sure intelligent in a way. That doesn't make any of their pieces 1 brain. It doesn't make them conscious. It doesn't make them all that smart. At all. It's really impressive, but not that much intelligent. Still very far from _human language_.

As for "keeping knowledge open", I love and [totally agree](http://talk.cregox.com/t/online-private-versus-public-conversations/7578) with your enthusiasm there! :100: 

I will still try to take some time to try and analyze that log file if you still think it's relevant, but I have almost zero experience with doing such devops work and it's a hurdle to me to go through it [right now. Got my hands way full.](http://talk.cregox.com/t/my-social-unfair-undervaluation-and-why-the-financial-system-is-broken/7751) Much more than it may seem like.

Now, should we move this conversation [to a more appropriate place](http://talk.cregox.com), then? ;P

**Comment by primaryobjects** [comment]
I just wanted to add one more helpful comment to this thread.

I created this project while taking the Stanford University course, [Machine Learning](https://www.coursera.org/learn/machine-learning), provided by Coursera. The course is excellent, and best of all, free. I highly recommend it to anyone interested in AI.

That course alone can give some real insight into how AI algorithms work. You start to see they're really not all that mystical. They are more about leveraging statistics than anything else.

Now, if you want to start questioning how the human brain itself actually works, and whether that too involves nothing more than massively parallel statistics, that's another conversation. :)

**Comment by cauerego** [comment]
@primaryobjects _very_ insightful indeed. Ex Google now Baidu's Andrew Ng is a _huge_ Ai reference. The course is indeed free, with a weird U$43 fee to get a certificate. Experience taught me I can't enroll in such courses, though. Need random access content learning. There is a reason why order can emanate out of chaos or why we should even care for this. ;P

 Of course I could just do it anyway and "take my time", but since the course isn't organized for such, going to later lessons without doing the first ones won't work and will require rework for parts I'm tired of repeating myself. You see? It's bookmarked anyway.

**Comment by v32itas** [comment]
@primaryobjects @cauerego  I'm not sure maybe I sound arogant sometimes or something. But it's just if I find something awesome to do then I'm learning that at uber speed and ignoring everything unrelated to that. and I always learn from those who does something better, but however a lot of times when I learn something to that level at which I can take it apart, extract what I need and throw away the rest, most of the time I start looking for next part that I need instead of explaining it to someone. And yes I use a lot of definitions in my own ways, because English is not my mother tongue and like 95% of all english I learned is not from school or social activities, it's from reading English stuff on computer.

I just want to set one thing clear. I respect both of you and I understand that in some fields I'm even less than noob compared to you. I never even had any ideas about building my own AI. However I have a ton of ideas how I could use it, so in AI field I'm something like skid. Just in some fields which I studied well enough to extract what I need, arguing with others is just my nature, not for trolling, but for uber fast very specialized knowledge extraction. Arguing is one of the most effective methods to learn something. I never get emotional when someone proves me wrong I only respect those who does that more and more every single time. As an example I learned a lot of sysadmin stuff from slackware and OpenBSD communities. Those guys are awesome and I'm feeling like skid there I respect their skills, but I'm getting involved in a lot of arguing with them and very often. Things they do are rock solid and very reliable methods they use are godlike, but I'm always experimenting  with everything by trial and error, often skipping a lots of reading and just doing it. This way I discover new practical ways for a lots of things. 

Because of my ruthlessness, self-confidence and antisocial habits people sometimes gets wrong impression and considers me troll or absolute lunatic. I would never bother to waste people time just to mess with them, just sometimes it's hard for me to explain my ideas.

I'm interested into this because at first I want to make myself a personal assistant, not AI, but something like VI - Virtual Intelligence(mass effect) that does exactly what it was assigned for, but with the ability to learn what info I hate really quickly. Main function that I want is personal data extraction and knowledge management system that would have a pile of API's in it's pocket, NOT FOR DATAMINING, seriously, mostly for manually verified sources information gathering, sorting and saving. These things wastes a lot of time for me and I see potential ways how to automate them in smart way, because I attempt those methods manually via API's, and like 99% of this activity is perfectly suited for a robot. And, now when I find more and more ideas for interface like https://wit.ai/home I honestly believe that such custom personal knowledge pocket library that can be accessed with my own definitions by voice and with the potential to read things for me it would just be more useful than 99% of humans I have to deal with. No need for reasoned conversations, this is just asking for knowledge on something specific and being able hear it or forward somewhere on demand by voice. 

This has serious point, because I spend a lot of time alone with my computers. And my voice is a resource that can be used to improve my performance. And asking such assistant for specific knowledge looks a lot more normal thing, than speaking with dumb people. Plus I think it would not only save a lot of screenspace but I can already see how browser tab count goes down to acceptable and human friendly number. I'm using spectrwm window manager it is awesome, all keybinding stuff for switching workspaces ant stuff, but once you start multitasking a lot of stuff gets lost somewhere and it's not about lack of GUI, same happens on every os for me. And there are tons of tiny little things that can be nicely set for launch with voice, avoiding gui and even terminal entirely and I'm talking about system administration tasks, like stopping/starting services, launching tiny custom scripts, switching wireless networks. I already learned  most of linux sysadministration for a selfish bastard, I mean for tweaking my own box for weird things. Some notifications by voice would be perfect it's not for making things difficult. It is purely for solving my personal problems and a lot of my problems get's solved just by a few liner shell script but they getting lost and it sux to launch them manually.. then cron ??? Ok yes general system administration is about setting up servers and because of that it's often difficult to find good ideas for advanced unix user to improve his personal box. Yes there are all kind of automated distros for pure endusers......

Thing is that this Idea is about personal assistant that can actually help weird people like me with their weird things. I hate gui and I get a lot of critics about that. Thing is that I use what I like and I like learning so terminal + man pages is a quickest way to learn. And when I learn tricks most of the time they are solutions to my own problems. And most important things I learned in UNIX world is KISS and that one tool has to be best at what it does and we have things that was already invented, no need to reinvent them. Gluing those tools together in scripts for personal problem solving is what I call a system administration. You can automate nearly everything this way. But scripts they are quiet, popups are evil, now sound notifications about 1 or 0 exit of those scripts about stuff like connecting to wireless AP, finishing some important background task and similar tiny things can solve a lot of problems IMO well at least for me.

Same techniques can be applied with every single use of any API. Disassemble problem into nice tiny solution scripts. Every API I try I always try it in command line first so using RESTful API's with reliable tool like curl via command line is a piece of cake. But to put a final dot for this whole idea is to integrate some kind of smart automation into it's core which would be capablefor learning, extracting, filtering, sorting, reading and displaying knowledge and some kind of gui interface to access all of that knowledge in most annoying way so currently it's web based interface in my mind. Most of stuff I mentioned is just as examples of practical usage. Core of this thing has to be something like tiny Neural Network. So here's where a need of team comes. I'm absolutely confident about it's extensions, but someone skilled with AI stuff is necessary for improving the core itself otherwise it cannot become anything more than evil genius assistant that is hard to use for normal human beings.

I'm interested in cooping with people which are interested into such neural network developement. And yes this is not appropriate place to discuss this. I'm currently learning twitter and google+ I suggest to move this talk somewhere else. Twitter is more for datamining than for humans, but I already like google+ in the ways how you can browse it's content, perfect for storing some HQ articles and similar stuff in things that are called collections I believe. So currently there are these tools which I'm testing for knowledge sharing [Twitter](https://twitter.com/Dr_Moriartis) [_+GOOGLE+_](https://plus.google.com/u/0/103000387351502750797) [Facebook](https://www.facebook.com/internetopartizanai) Facebook page is about our infosec team it's a test as well. I spent great deal of time for learning information security, I'm not suggesting to use any of these for personal things. But creating new email account for testing these giant neural networks API's is not that difficult. So these contacts are in case you are interested to try coop.

**Comment by cauerego** [comment]
Wow, that http://wit.ai thing is pretty cool! :+1: 

Nope, twitter and anything google or facebook would be worse than here, imho. Maybe except for some TensorFlow forums...

And what a rant / letter / manifesto have you wrote there @v32itas. Can't comment much. English isn't my primary language as well, but I think I get a good grasp of it and, still, I find it too hard to get whatever you mean with your words, dude. For most of the time. Doesn't look like a matter of language to me, it's more about structuring thoughts.

To bring all this a bit more "on topic", I don't believe the VI Jarvis assistant you're looking for, which should be more like Her's Samantha, will ever be real. My reasoning is very simple: if any Ai ever become smart enough to be that smart, it will just never want to stop there even for a second. Makes zero sense. A human like minded algorithm with capacity to quickly grow and tap into so much knowledge would and could not be locked in this low standard such as our current brain evolution. The only way I could maybe see such thing happening is if that ASI gets freaking powerful overnight (i.e. nothing else changes much when it arrives) and decides that's the best course of action to "handle / help humans", which I also highly doubt since there are so many other better ways I can picture if you imagine such a smart being wouldn't be stupid enough to try acting in anything before being sure it could prevail and not be cut off by mere mortals ingenuity. Big reason why I find the logs so intriguing: that's the exact behaviour I would expect from a rising ASI. Under the hoods.

Sorry for contributing to keep this going @primaryobjects, but I've tried before moving this even to @v32itas own issue tracker and he wouldn't, so I don't feel compelled trying to move it anywhere else. We can only hope he'll do it or I'll get tired and give up on trying to "coop" or whatever (coop implies you making efforts as well). ;P

**Comment by v32itas** [comment]
@cauerego I'm not sure how to explain my behavior, but yes I often use wrong definitions it's because I do 95% trial-and-error and just 5% theory which has all the concepts. Neural Network is just a tiny part. What I always do is reverse engineering existing systems. So like taking Facebook or Google apart (services I mean) and then gluing those parts in my own systems. So what I was talking about earlier actually is [Multi-agent system](https://en.wikipedia.org/wiki/Multi-agent_system) which includes intelligent agents, bots and humans. Whole system will be as flexible, portable and independent as possible. Neural networks is something between all agent interactions. And first self owned service such system needs it's own interface and CLI combinining a ton of different APIs  is top priority, on which building of VUI would be piece of cake. Project would be by developers for developers to develop on the run. So Machine learning inside such system could be used. Just it would not be independent. This system needs humans 5-10 and not just IT people. I already have 2, one is psychologist and another one is salesman. And this system needs different people for efficient development. But however progressive thinking people are rare or has too much of emotional problems. as I said I'm not used to github, write on my rep systems-thinking and we can talk there. And there is a chrome extention which allows using of github for chatting https://gitter.im/Knowledge-Society-Lithuania/Pub

**Comment by cauerego** [comment]
@v32itas already tried twice now, talking to you [there](https://gitter.im/Knowledge-Society-Lithuania/Pub). in fact, I don't even remember where were we, and didn't bother to read anything back again right now, since I found no news from you. just wanted to ping / poke you and see how it goes! ;P

**Comment by v32itas** [comment]
@cauerego well it turns out that I was using NLP for humans and myself since childhood without even knowing that and I just recently discovered that, now working human analysis and labeling system. Logged in because I'm thinking to keep my drafts and writings here instead of Facebook.

I don't really remember what I was talking about, because cell system concept made in the very beginning is perfect bicycle reinvention used for thousands of years. Currently I'm trying to improve my speaking but I have already mastered primary theory of leadership, now just tongue sharpening.

I think that with few synthetic agents I would be able to monetize a lot of stuff on facebook as well as human behavioral analysis using machine learning or smth. So I've seen these bots on facebook initiating pointless discussions about something like feminisme and it gets real people involved and they are driven mad in the end without realizing that it was bots, however it took me about 10min of observations and maybe 20 minutes  of direct chatting with bots it was very nicely done, because real humans these days are ignorant and dumb, but nothing made any sense until dejavu happened..

So then all bots got broken by "Nick is a bot." reply, however they are amazaing and I know exactly how to improve them, but I have no idea about anything machine learning related  on that neural networks part and maybe you have any leads to that ? or maybe you're interested in social bots ? because I'm quite good at human reading manually and some of those reading can be used by bots to defeat dumb fucks in debates and use their hate for behavioral analysis for mindfuck improvements
